internal/blocksync/reactor.go

```diff
@@ -443,6 +443,8 @@ func (r *Reactor) poolRoutine(ctx context.Context, stateSynced bool, blockSyncCh
 		lastRate    = 0.0
 
 		didProcessCh = make(chan struct{}, 1)
+
+		initialCommitHasExtensions = (r.initialState.LastBlockHeight > 0 && r.store.LoadBlockExtendedCommit(r.initialState.LastBlockHeight) != nil)
```

Comment:
Suggestion: if you take the "read the store" approach (i.e. `r.store.LoadBlockExtendedCommit(r.initialState.LastBlockHeight) != nil`),
then we can simplify the condition (I think, please keep me honest!). It could look like this:
``` go
    case state.ConsensusParams.ABCI.VoteExtensionsEnabled(state.LastBlockHeight) &&
        r.store.LoadBlockExtendedCommit(r.initialState.LastBlockHeight) == nil:
```

Reply:
After offline conversation, you can ignore this. I get it now